### PR TITLE
[RDY] Fix performance of cluster query.

### PR DESCRIPTION
**Pull request by gfokkema**
Two fixes have been provided,
Please comment on whether to use:
- 225541c
Only clusters to the right, with the downside that the start rank is never clustered.
Therefore edges do some weird stuff at the start.  
Upside: is faster.
- acd7787
Clusters to left and right, but does not return or pursue clusters that start before the clusters of the start ranks.  
Upside: possibly more correct.  
Downside: possibly still clusters a huge part of the graph.  
_Correction: since the startRank of the initial clusters is equal to the rank of their startnodes, this downside does not apply._

### CODE REVIEW

**Comment by pvanagtmaal** [comment]
I'd prefer the second solution, [acd7787](https://github.com/DNAinator/dnainator/commit/acd778762fab96a87c6a90983ac3d500769fb738)

**Comment by Unia** [comment]
How much faster is the first solution? Is the difference in speed noticable and if so, how much?

File: src/main/java/nl/tudelft/dnainator/graph/impl/query/AllClustersQuery.java [hunk 1]
```diff
@@ -50,12 +50,12 @@ public AllClustersQuery(List<String> startNodes, int maxRank, int threshold) {
 		rootClusters.addAll(clusterFrom.execute(service));
 
 		// Find adjacent clusters as long as there are root clusters in the queue
+		int minrank = rootClusters.stream().mapToInt(e -> e.getStartRank()).min().getAsInt();
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/graph/impl/query/AllClustersQuery.java]: Throws an exception when scrolling further than the end of the graph.

**Comment by Balletie** [comment]
Alright, merging.

**Pull request merged**